### PR TITLE
Arc submissions

### PR DESCRIPTION
**Pull request by armstrongnate**
refs: MBL-10231
affects: student
release note: none

### CODE REVIEW

File: Core/Core/Database/CoreData/Models/Submission.swift [hunk 1]
```diff
@@ -199,7 +199,7 @@ extension Submission {
     public var subtitle: String? {
         guard let type = type else { return nil }
         switch type {
-        case .basic_lti_launch, .external_tool, .online_quiz:
+        case .basic_lti_launch, .external_tool, .online_quiz, .arc:
```
Comment (vanwagonet) [inline Core/Core/Database/CoreData/Models/Submission.swift]: I didn't realize they got their own submission type.

File: Student/Student/Submissions/ArcSubmission/ArcSubmissionViewController.swift [hunk 2]
```diff
@@ -49,8 +49,27 @@ class ArcSubmissionViewController: UIViewController, ArcSubmissionView {
...
-        webView.navigationDelegate = self
         spinner.startAnimating()
+
+        let config = WKWebViewConfiguration()
+        config.processPool = CoreWebView.processPool
+        webView = WKWebView(frame: .zero, configuration: config)
+        webView.navigationDelegate = self
+        webView.addScript(js)
+        webView.handle("submit") { [weak self] message in
+            self?.presenter?.submit(form: message.body) { error in
```
Comment (vanwagonet) [inline Student/Student/Submissions/ArcSubmission/ArcSubmissionViewController.swift]: `message.body as? [String: Any]` should work. You shouldn't need to use `JSONSerilaization` to get a dict representation of the message.

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Please run `yarn update-headers`. The following do not have the correct license header: 
* Core/CoreTests/Extensions/WKWebViewExtensionsTests.swift
* Core/CoreTests/Use Cases/GetArcTests.swift
* Student/StudentUnitTests/Submissions/ArcSubmission/ArcSubmissionPresenterTests.swift
* TestsFoundation/TestsFoundation/Fixtures/API/APIExternalToolFixture.swift
* TestsFoundation/TestsFoundation/Fixtures/Model/ExternalToolFixture.swift
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Student

[MBL-10231](https://instructure.atlassian.net/browse/MBL-10231)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Core** | 95.1% | 95.08% | 0.02%
**Student** | 92.87% | 92.34% | 0.53%
**Teacher** | 100% | 100% | 0%
**React Native** | 92.71% | 92.71% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Student/Student/Submissions/ArcSubmission/ArcSubmissionViewController.swift (hunk 2 above)
Comment (armstrongnate) [reply Student/Student/Submissions/ArcSubmission/ArcSubmissionViewController.swift]: I refactored this.

File: Core/Core/Database/CoreData/Models/Submission.swift (hunk 1 above)
Comment (armstrongnate) [reply Core/Core/Database/CoreData/Models/Submission.swift]: It's not a submission type from the api. I created it and add it to the list of types if the course has arc.

Comment (vanwagonet) [reply Core/Core/Database/CoreData/Models/Submission.swift]: I think adding it to the submission types will be confusing in the long run. From multiple places you'd get the impression that it is coming from the api, and that this is the way we know it's an Arc submission.

**Pull request merged**